homeassistant/components/sms/notify.py

```diff
@@ -42,7 +42,15 @@ async def async_send_message(self, message="", **kwargs):
             _LOGGER.error("No target number specified, cannot send message")
             return
 
-        is_unicode = kwargs.get(CONF_UNICODE, True)
+        extended_data = kwargs.get(ATTR_DATA, {})
+        _LOGGER.debug("Extended data:%s", extended_data)
+
+        if extended_data is None:
```

Comment:
extended_data cant be None here, since you default it to {} so this check is not needed.

Reply:
If I dont check, the code in line 51 will fail with missing key

Reply:
extended_data is a always a dict after the get(attr_data, {}). Which you then call get(conf_unicode, True) willl return True if attribute is missing.

Reply:
This is what I get: 
```
2022-09-20 14:50:05.947 ERROR (MainThread) [homeassistant.helpers.script.websocket_api_script] websocket_api script: Error executing script. Unexpected error for call_service at pos 1: 'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "/home/oscar/repos/core/homeassistant/helpers/script.py", line 450, in _async_step
    await getattr(self, handler)()
  File "/home/oscar/repos/core/homeassistant/helpers/script.py", line 683, in _async_call_service_step
    await service_task
  File "/home/oscar/repos/core/homeassistant/core.py", line 1738, in async_call
    task.result()
  File "/home/oscar/repos/core/homeassistant/core.py", line 1775, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/home/oscar/repos/core/homeassistant/components/notify/legacy.py", line 258, in _async_notify_message_service
    await self.async_send_message(**kwargs)
  File "/home/oscar/repos/core/homeassistant/components/sms/notify.py", line 48, in async_send_message
    is_unicode = extended_data.get(CONF_UNICODE, True)
AttributeError: 'NoneType' object has no attribute 'get'
```

Reply:
Okey, seems a strangeness with the service definition then. I suggest you change the get(attr_data, {}) to just get(attr_data) to make it less confusing.

Reply:
done